config.py

```
@@ -0,0 +1,418 @@
+- Environment variable overrides
+- Configuration file loading (YAML/JSON)
+- Sensible defaults
+- Type validation and error handling
+- Hierarchical configuration management
+"""
+
+import os
+import yaml
+import json
+from pathlib import Path
+from typing import Dict, Any, Optional, Union
```

Ruff (0.12.2)

`typing.Dict` is deprecated, use `dict` instead

(UP035)